### PR TITLE
DOC: Improve the denoising workflow documentation

### PR DESCRIPTION
**Pull request by jhlegarreta**
Improve the denoising workflow documentation:
- Move general comments about PCA to the place where it is introduced for the
  first time.
- Improve the explanations about the command arguments.
- Fix grammar.
- Use a given line length consistently.

### CODE REVIEW

**Comment by jhlegarreta** [comment]
Making sure this is consistent with the review that @arokem and @ShreyasFadnavis did on the blog post document.

**Comment by ShreyasFadnavis** [comment]
Hi @jhlegarreta ! Looks good -- 1 minor change in general: we should probably stick to `Overcomplete Local PCA` and `Marchenko-Pastur PCA`  (both of which are variants of the local PCA method first (AFAIK) introduced for DWIs here: https://onlinelibrary.wiley.com/doi/full/10.1002/mrm.22748)

**Comment by ShreyasFadnavis** [comment]
> Hi @jhlegarreta ! Looks good -- 1 minor change in general: we should probably stick to `Overcomplete Local PCA` and `Marchenko-Pastur PCA` (both of which are variants of the local PCA method first (AFAIK) introduced for DWIs here: https://onlinelibrary.wiley.com/doi/full/10.1002/mrm.22748)

This can be addressed on a later PR of course!

**Comment by jhlegarreta** [comment]
> Hi @jhlegarreta ! Looks good -- 1 minor change in general: we should probably stick to `Overcomplete Local PCA` and `Marchenko-Pastur PCA` (both of which are variants of the local PCA method first (AFAIK) introduced for DWIs here: https://onlinelibrary.wiley.com/doi/full/10.1002/mrm.22748)

Thanks for the review @ShreyasFadnavis. Done in commint 3407c52.

**Comment by jhlegarreta** [comment]
Build failures are unrelated.

**Review by areeshatariq — APPROVED** [review]

File: doc/interfaces/denoise_flow.rst [hunk 1]
```diff
@@ -10,43 +10,53 @@ Multiple denoising methods are available in DIPY.
...
-The algorithm to perform the PCA-based denoising involves the following steps:
+The algorithm to perform an Overcomplete Local PCA-based (LPCA) denoising
+involves the following steps:
 
-* First, we estimate the local noise variance at each voxel.
+* Estimating the local noise variance at each voxel.
+* Applying a PCA in local patches around each voxel over the gradient
+directions.
+* Thresholding the eigenvalues based on the local estimate of the noise
+variance, and then doing a PCA reconstruction.
```
Comment (skoudoro) [inline doc/interfaces/denoise_flow.rst:30]: As you can see below, there is an issue with the rendering. Can you fix it

Capture d’écran 2020-11-10 à 23 24 25

**Review by skoudoro — COMMENTED** [review]
Hi @jhlegarreta,

Thank you for this. When I generate this tutorial, I have the following error:

> :scale:option is ignored

Could you fix it too?

Thanks

**Comment by skoudoro** [comment]
To generate the doc and see the result:   `make -C doc html-after-examples`

File: doc/interfaces/denoise_flow.rst (hunk 1 above)
Comment (jhlegarreta) [reply doc/interfaces/denoise_flow.rst]: Good catch. Sorry. Fixed in ca07e4a.

**Comment by jhlegarreta** [comment]
> Thank you for this. When I generate this tutorial, I have the following error:
> :scale:option is ignored
> Could you fix it too?
> Thanks

I assume that this warning/error was already present before this PR. It looks like Sphinx needs `Pillow` for the `scale` option to work. The [`doc-requirements.txt`](https://github.com/dipy/dipy/blob/master/doc/doc-requirements.txt) does not list it.

> To generate the doc and see the result: make -C doc html-after-examples

Thanks for the pointer :+1:. I believe the corresponding CI build should put in place to free developers from the burden of doing that additional step.

**Comment by skoudoro** [comment]
> I assume that this warning/error was already present before this PR. It looks like Sphinx needs Pillow for the scale option to work. The doc-requirements.txt does not list it.

`Pillow` is installed in my environment, I do not think this is the reason. Could you fix this issue here or you prefer a dedicated PR? 

> Thanks for the pointer 👍. I believe the corresponding CI build should put in place to free developers from the burden of doing that additional step.

Like writing tests, I think it is a good practice to generate the documentation locally to check your change before pushing or creating a PR. Although I agree we should have a dedicated CI's, I think the developer should always check his work.

**Comment by jhlegarreta** [comment]
For the time being, maybe the `scale` issue can be addressed in a separate PR as it is a warning I believe. Thanks for having checked the first version of the generated documentation.

**Comment by skoudoro** [comment]
After some thinking, the issue should be fixed on another PR.  Thank you for this @jhlegarreta. Merging

**Pull request merged**